internal/delete_handler.go

```diff
@@ -73,26 +73,38 @@ func getBackupsAndGarbage(folder storage.Folder) (backups []BackupTime, garbage
 		return nil, nil, err
 	}
 
-	sortTimes := getBackupTimeSlices(backupObjects)
+	sortTimes := getBackupTimeSlices(backupObjects, folder)
 	garbage = getGarbageFromPrefix(subFolders, sortTimes)
 
 	return sortTimes, garbage, nil
 }
 
 // TODO : unit tests
-func getBackupTimeSlices(backups []storage.Object) []BackupTime {
-	sortTimes := make([]BackupTime, len(backups))
-	for i, object := range backups {
+func getBackupTimeSlices(backups []storage.Object, folder storage.Folder) []BackupTime {
+	sortTimes := make([]BackupTime, 0)
+	for _, object := range backups {
 		key := object.GetName()
 		if !strings.HasSuffix(key, utility.SentinelSuffix) {
 			continue
 		}
-		time := object.GetLastModified()
-		sortTimes[i] = BackupTime{utility.StripBackupName(key), time, utility.StripWalFileName(key)}
+		name := utility.StripBackupName(key)
+		backup, err := GetBackupByName(name, folder)
+		if err != nil {
+			continue
+		}
+		var finishTime time.Time
+		meta, err := backup.FetchMeta()
```

Comment:
Some users have huge amount of backups. So this type of listing could create high-load reading for storage.
We think it would be better to use this type of listing only with explicit flag --details.